WFLY-19995 Upgrade to Hibernate Search 7.2.2.Final / WFLY-19996 Upgrade to Elasticsearch client 8.15.4 / WFLY-19997 Upgrade to com.carrotsearch:hppc 0.10.0 / WFLY-20013 Upgrade to Hibernate ORM 6.6.3.Final

marko-bekhta opened the pull request:
https://issues.redhat.com/browse/WFLY-19995
https://issues.redhat.com/browse/WFLY-19996
https://issues.redhat.com/browse/WFLY-19997
https://issues.redhat.com/browse/WFLY-20013

Hibernate Search 7.2.2 brings a few bug fixes and dependency updates. 

I noticed that the last time we did a Lucene upgrade, I completely missed the hppc dependency, hence the changes in the Lucene module. 

While Hibernate Search also bumped Hibernate ORM to 6.6.3, I'm leaving out that update from this PR as I think @scottmarlow was planning on doing some other changes related to bytecode enhancement.

marko-bekhta committed:
[WFLY-19995] Upgrade to Hibernate Search 7.2.2.Final

marko-bekhta committed:
[WFLY-19996] Upgrade to Elasticsearch client 8.15.4

marko-bekhta committed:
[WFLY-19996] Use Elasticsearch 8.15.4 in tests

marko-bekhta committed:
[WFLY-19997] Upgrade to com.carrotsearch:hppc 0.10.0

marko-bekhta committed:
[WFLY-19997] Remove com.carrotsearch:hppc from the Lucene module

- since Lucene starting with 9.11 comes its own internal version of hppc packages as part of the Lucene itself

scottmarlow commented:
> While Hibernate Search also bumped Hibernate ORM to 6.6.3, I'm leaving out that update from this PR as I think @scottmarlow was planning on doing some other changes related to bytecode enhancement.

https://github.com/wildfly/wildfly/pull/18313 is the additional change that I would like to merge before this pull request or perhaps together with this pull request.  

I did test locally with some of the 6.6.3 changes but will run more tests.

bstansberry commented:
Note that the 'Linux with security manager' CI job will fail on a number of fault-tolerance / micrometer tests. Ignore those failures (i.e. no need to re-run). They are due to a known issue addressed by #18428

bstansberry commented:
@scottmarlow I merged #18313. Please let me know what the plan is re this one, i.e. please approve if it can go in as-is or if you want, submit a PR based on this branch but with an added commit to upgrade ORM.

marko-bekhta committed:
[WFLY-20013] Upgrade to Hibernate ORM 6.6.3.Final

marko-bekhta commented:
Hey @bstansberry , @scottmarlow  😃 👋🏻 

I created a Jira and added a commit to bump ORM to 6.6.3 (also rebased to include the change for bytecode enhancement). Let's see what CI will say now 🙈 🤞🏻

scottmarlow commented:
> @scottmarlow I merged #18313. Please let me know what the plan is re this one, i.e. please approve if it can go in as-is or if you want, submit a PR based on this branch but with an added commit to upgrade ORM.

+1 for merging this.  

I see that https://issues.redhat.com/browse/WFLY-20013 is for upgrade to Hibernate ORM 6.6.3.Final which is now a commit in this pull request as well.

@marko-bekhta could you please edit the description to include the link to https://issues.redhat.com/browse/WFLY-20013 as well.

bstansberry commented:
Thanks @marko-bekhta and @scottmarlow
